rasa/core/brokers/kafka.py

```diff
@@ -124,67 +124,73 @@ def publish(
                     f"Could not publish message to kafka url '{self.url}'. "
                     f"Failed with error: {e}"
                 )
-                connected = self.producer.bootstrap_connected()
-                if not connected:
-                    self._close()
+                try:
+                    self.producer.flush(timeout=1)
+                except KafkaException:
                     logger.debug("Connection to kafka lost, reconnecting...")
                     self.producer = self._create_producer()
-                    connected = self.producer.bootstrap_connected()
-                    if connected:
+                    try:
+                        self.producer.flush(timeout=1)
                         logger.debug("Reconnection to kafka successful")
                         self._publish(event)
+                    except KafkaException:
+                        pass
                 retries -= 1
                 time.sleep(retry_delay_in_seconds)
 
         logger.error("Failed to publish Kafka event.")
 
-    def _create_producer(self) -> "KafkaProducer":
-        import kafka
+    def _create_producer(self) -> "SerializingProducer":
+        import confluent_kafka
 
         if self.security_protocol == "PLAINTEXT":
-            authentication_params = dict(
-                security_protocol=self.security_protocol, ssl_check_hostname=False
-            )
+            authentication_params: Dict[Text, Any] = {
+                "security.protocol": self.security_protocol.lower(),
+            }
         elif self.security_protocol == "SASL_PLAINTEXT":
-            authentication_params = dict(
-                sasl_plain_username=self.sasl_username,
-                sasl_plain_password=self.sasl_password,
-                sasl_mechanism=self.sasl_mechanism,
-                security_protocol=self.security_protocol,
-            )
+            authentication_params = {
+                "sasl.username": self.sasl_username,
+                "sasl.password": self.sasl_password,
+                "sasl.mechanism": self.sasl_mechanism,
+                "security.protocol": self.security_protocol.lower(),
```

Comment:
is the lower needed? in the confluent examples it isn't there https://github.com/confluentinc/confluent-kafka-python/blob/master/examples/sasl_producer.py#L59

Reply:
I think I was getting some errors without the lower part, I'll double-check again.

Reply:
Removed the lower(): https://github.com/RasaHQ/rasa/pull/11765/commits/e8f9aa8f489e02d4f14a337160d70405fa118321